src/adventure/xpstatuswidget.cpp

```diff
@@ -0,0 +1,66 @@
+#include "xpstatuswidget.h"
+#include "adventuresession.h"
+#include "adventuretracker.h"
+#include "adventurewidget.h"
+
+XPStatusWidget::XPStatusWidget(AdventureTracker &at, QStatusBar *sb, QWidget *parent)
+    : QPushButton(parent)
+    , m_statusBar{sb}
+    , m_tracker{at}
+    , m_session{}
+{
+    setFlat(true);
+    setStyleSheet("QPushButton { border: none; outline: none; }");
+
+    setMouseTracking(true);
+
+    updateContent();
+
+    connect(&m_tracker,
```

Comment:
AdventureTracker can be on a different thread. I wasn't mindful as part of the review around where we might run into race conditions.

Reply:
Yeah, I've been thinking about this a lot (although not sure those thoughts accomplish much, lol). But in this case (for a UI widget consumer like XPStatus):
- the Tracker only emits a const reference
- the Tracker does not overwrite / rebuild in place when a new session starts, it allocates a new object when a new session starts (previous versions did rebuild in place and I was worried that would cause subtle bugs)
- The UI widget makes a deep copy and only interprets / renders based on the deep copy

Does this seem reasonable?